inst/include/conditionalSampler.h

```diff
@@ -0,0 +1,673 @@
+            ///Returns the n'th element of the reference trajectory
+            Space GetReferenceValue(long n) const {return referenceTrajectory[n];}
+            ///Returns a constant reference to the n'th element of the reference trajectory
+            const Space & GetReferenceValueRefs(long n) const {return referenceTrajectory[n];}
+            ///Returns index of conditional reference Trajectory
+            const long GetReferenceValueIndex(long lTime) const {return(referenceTrajectoryIndices.at(lTime));
+            }
+            ///Sets the reference trajectory
+            void SetReferenceTrajectory(const std::vector<Space>& newReferenceTrajectory) {referenceTrajectory = newReferenceTrajectory;}
+            ///Sets the n'elment of the reference trajectory
+            void SetReferenceValue(const Space & newReferenceValue, long n) {referenceTrajectory[n] = newReferenceValue;}
+            ///Initialise the conditional sampler and its constituent particles. The reference trajectory used to set the first coordinate at T=0 is either from the last call to conditionalSampler.SetReferenceTrajectory(), or defaults to the trajectory passed to the constructor at the time the object is created.
+            void Initialise(void);
+            ///Initialise the sampler and its constituent particles.
+            void Initialise(const std::vector<Space>& newReferenceTrajectory);
+            ///Perform one conditional iteration of the simulation algorithm.
+            void Iterate(void);
+            ///Perform one conditional iteration of the simulation algorithm and return the resulting ess
+            double IterateEss(void);
+            ///Perform conditional iterations until the specified evolution time is reached
+            void IterateUntil(long lTerminate);
+            ///Resample the particle set using the specified resampling scheme specifically adjusted for conditional resampling
+            void conditionalResample(ResampleType::Enum lMode);
+            ///Move the reference particle by setting the coordinate to the reference value and re-weighting.
+            void MoveReferenceParticle(void);
+            ///Dump a specified particle to the specified output stream in a human readable form
+            std::ostream & StreamParticle(std::ostream & os, long n) const;
+            ///Dump the entire particle set to the specified output stream in a human readable form
+            std::ostream & StreamParticles(std::ostream & os, int digits) const;
+            friend std::ostream& operator << <>(std::ostream &, const conditionalSampler<Space,Params>&);
+
+            ///Throws exception when adaptation related members of the base sampler class are used:
+            void SetAdaptMethods(adaptMethods<Space,Params>* adaptMethod) {throw SMC_EXCEPTION(CSMCX_USING_ADAPTATION, "Adaptation methods not supported for conditional sampler class.");}
```

Comment:
I'd personally be inclined to add the word currently before supported in these messages; in a sense most of these are things we haven't implemented because they're complicated and time is precious rather than because we've made a conscious decision not to.

Reply:
Done! But I changed the description of the conditionalSampler class line l. 25 of `conditionalSampler.h` to provide a bit more information and documentation for the user. 

It's probably best if you have a look at the exact wording of this line to make that statement as precise and clear as it needs to be (it's a language thing and my english is clearly only my second best :) )